index.js

```diff
@@ -80,6 +80,14 @@ function _baseParse (base) {
 
   return resultObject;
 }
+
+function _isAbsolute(relative) {
+  // https://tools.ietf.org/html/rfc3986#section-3.1
+  const scheme = '[a-z][a-z0-9+.-]*'; // ALPHA *( ALPHA / DIGIT / "+" / "-" / "."  )]
+  const isAbsolutePattern = new RegExp(`^(${scheme}:)?//`, 'i');
```

Comment:
This regex looks static, we could hoist it outside of this function.

Reply:
Good call! Fixed.